### PR TITLE
[`SAM`] Fixes pipeline and adds a dummy pipeline test

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

1- Fixes the automatic mask generation pipeline, currently on the main branch, the script below

```python
from transformers import pipeline
from PIL import Image
import requests

generator = pipeline("mask-generation", model="facebook/sam-vit-base", device=0)

img_url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/transformers/tasks/car.jpg"
raw_image = Image.open(requests.get(img_url, stream=True).raw).convert("RGB")

outputs = generator(raw_image, points_per_batch=64)
```
is broken

2- Adds a dummy pipeline test. I know the pipelines are already tested in tests/pipeline but these tests are quite slow. I thought adding a small dummy pipeline test is easier for future contributors to make sure they will not break the pipeline without having to run the entire pipeline testing suite for SAM

cc @ArthurZucker @ydshieh @sgugger

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
LGTM, test could be `test_dummy_pipeline_generation` since we don't check the results

File: src/transformers/models/sam/image_processing_sam.py [hunk 1]
```diff
@@ -934,7 +934,7 @@ def _generate_crop_boxes(
     cropped_images, point_grid_per_crop = _generate_crop_images(
         crop_boxes, image, points_grid, layer_idxs, target_size, original_size
     )
-
+    crop_boxes = np.array(crop_boxes)
```
Comment (ydshieh) [inline src/transformers/models/sam/image_processing_sam.py:937]: when it currently fails, what is the type of `crop_boxes`? Is it a torch tensor? If so, we can't rely on `np.array` unless we are 100% sure `crop_boxes` is never on cuda.

Comment (ArthurZucker) [reply src/transformers/models/sam/image_processing_sam.py:937]: Pre-processing should never be done on cuda no?

File: tests/models/sam/test_modeling_sam.py [hunk 2]
```diff
@@ -751,3 +751,9 @@ def test_inference_mask_generation_three_boxes_point_batch(self):
         iou_scores = outputs.iou_scores.cpu()
         self.assertTrue(iou_scores.shape == (1, 3, 3))
         torch.testing.assert_allclose(iou_scores, EXPECTED_IOU, atol=1e-4, rtol=1e-4)
+
+    def test_dummy_pipeline_generation(self):
+        generator = pipeline("mask-generation", model="facebook/sam-vit-base", device=torch_device)
+        raw_image = prepare_image()
+
+        _ = generator(raw_image, points_per_batch=64)
```
Comment (ydshieh) [inline tests/models/sam/test_modeling_sam.py:759]: I am surprised that this is not decorated by `require_vision`, but even more surprised that this test file has no `require_vision` at all, but having 

```python
if is_vision_available():
    from PIL import Image
```

This has to be fixed, but it's OK if we do this in a separate PR and maybe delegate the task to the original SAM modeling author in `transformers`.

**Review by ydshieh — APPROVED** [review]
LGTM after @ArthurZucker's explanation. Just a remakr on the usage of `require_vision`.

**Review by sgugger — APPROVED** [review]
Thanks for the fixes!
